app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt

```diff
@@ -25,25 +25,49 @@ import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.github.se.travelpouch.R
+import com.github.se.travelpouch.model.profile.CurrentProfile
+import com.github.se.travelpouch.model.profile.ProfileModelView
+import com.github.se.travelpouch.model.travels.ListTravelViewModel
 import com.github.se.travelpouch.ui.navigation.NavigationActions
 import com.github.se.travelpouch.ui.navigation.Screen
 import com.google.android.gms.auth.api.signin.GoogleSignIn
 import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 
 /**
  * A composable function that displays the sign-in screen.
  *
  * @param navigationActions An instance of `NavigationActions` to handle navigation events.
  */
+@OptIn(DelicateCoroutinesApi::class)
 @Composable
-fun SignInScreen(navigationActions: NavigationActions) {
+fun SignInScreen(
+    navigationActions: NavigationActions,
+    profileModelView: ProfileModelView,
+    travelViewModel: ListTravelViewModel
+) {
   val context = LocalContext.current
 
   // launcher for Firebase authentication
   val launcher =
       rememberFirebaseAuthLauncher(
           onAuthComplete = { result ->
             Log.d("SignInScreen", "User signed in: ${result.user?.displayName}")
+            val job =
+                GlobalScope.launch {
+                  profileModelView.initAfterLogin()
```

Comment:
I will dig into that. I am not sure of what is does. Thbe problem is that I cannot call the travelViewModel.initAfterLogin() to fetch the profile since I have to first know the uid of the user connected.

Reply:
Yes, I corrected the blocking thread. I did not need it, the init functions with AuthListeners were sufficient